src/data/set/basic.lean

```diff
@@ -1112,6 +1184,10 @@ set.ext $ assume a,
 @[simp] lemma val_range {p : α → Prop} :
   set.range (@subtype.val _ p) = {x | p x} :=
 by rw ← set.image_univ; simp [-set.image_univ, val_image]
+
+@[simp] lemma range_val (s : set α) : range (subtype.val : s → α) = s :=
```

Comment:
I'm not happy with the distinction between `range_val` and `val_range`. It looks inconsistent. Also this one should use `coe` instead of `subtype.val`

Reply:
do you have a suggestion for the naming?

Reply:
I think `val_range` should be renamed to `range_val`, and your `range_val` should be renamed `range_set_coe`